lib/execution_engine2/utils/catalog_cache.py

```diff
@@ -0,0 +1,89 @@
+from collections import defaultdict
+from typing import Dict
+
+from lib.installed_clients.CatalogClient import Catalog
+
+
+class CatalogCache:
+    """
+    Per call catalog cache used to speed up catalog lookups
+    Caches the "Method Version" and the "Job Resource Requirements"
+    """
+
+    def __init__(self, catalog: Catalog):
+        if not catalog:
+            raise ValueError("Please provide instance of catalog client")
+
+        self._catalog = catalog
+        self._method_version_cache = defaultdict(dict)
+        self._job_requirements_cache = defaultdict(dict)
+
+    def get_catalog(self) -> Catalog:
```

Comment:
This and the next 2 method appear to be untested

Reply:
How would you test a getter?

catalog = Catalog()
catalog_cache =  CatalogCache(catalog)
assert catalog == CatalogCache.get_catalog() ?

Reply:
Pretty much, it's generally super simple.

Reply:
done

Reply:
For the cache getters, I was thinking more along the lines of running a couple lookups and then checking the cache contents are as you expect

Reply:
Ok, so removed this tests and added assertions on the cache contents in the two other tests in this file

Reply:
The test for `get_catalog` was fine as is, it was just the cache getters I was talking about

Reply:
:+1: